proto/svc-storage-grpc-parcel.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+package grpc.parcel;
+
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "svc-storage-grpc.proto";
+
+
+// Response struct returning an [Object] on success and [ValidationResult] if invalid fields were provided
+message Response {
+    // struct with field -> error pairs to provide feedback about invalid fields
+    grpc.ValidationResult validation_result = 1;
+    // Object struct with id [String] in [Uuid](uuid::Uuid) format and [Data] struct with parcel data
+    optional Object object = 2;
+}
+
+// Object struct with `id` and `data` field
+// * `id` [String] in [Uuid](uuid::Uuid) format
+// * `data` [Data] struct with parcel data
+message Object {
+    //id UUID v4
+    string id = 1;
+    //data
+    Data data = 2;
+}
+
+// UpdateObject struct with `id`, `data` and `mask` fields
+// * `id` [String] in [Uuid](uuid::Uuid) format
+// * `data` [Data] struct with parcel data which should be used for update
+// * `mask` [FieldMask] struct with parcel fields that should be updated
+message UpdateObject {
+    // `id` [String] in [Uuid](uuid::Uuid) format
+    string id = 1;
+    // struct with parcel data which should be used for update
+    Data data = 2;
+    // struct with parcel fields that should be updated
+    google.protobuf.FieldMask mask = 3;
+}
+
+// Parcel Status Enum
+enum ParcelStatus {
+    // Has not been dropped off
+    NOTDROPPEDOF = 0;
+    // Has been dropped off
+    DROPPEDOFF = 1;
+    // En Route (could be a number of vertiport hops)
+    ENROUTE = 2;
+    // Arrived at destination port
+    ARRIVED = 3;
+    // Picked up by recipient
+    PICKEDUP = 4;
+    // Transaction complete & parties have been notified of the
+    // completed transaction
+    COMPLETE = 5;
+}
+
+// Data struct with parcel data
+message Data {
+    // the itinerary UUID associated with the parcel
+    string itinerary_id = 1;
+
+    // the UUID represents the current owner of the parcel. It could be
+    // a vertiport, a vehicle or a user, depending on the current status.
+    string owner_id = 2;
```

Comment:
On a rethink of this, this might not be necessary

The (eventual) package_scans table will have the following columns:
- package id
- scanned_at (indexed) time
- scanner_id (user_id or vehicle_id) can be employee, vehicle, or customer
- location_id (vertiport, etc.)

When a scan comes into /api/cargo/package/scan/, the handler can update the parcel status based on the information provided, compared to the package's itinerary

Likewise when someone requests an update of their package, we will want to probably grab ALL of the scans from the table and display them (chain of custody of package), so copying the current owner to its own field might not make sense

![image](https://github.com/Arrow-air/svc-storage/assets/6334638/be2d7311-49f7-4a14-9063-36517a2ac58b)

Reply:
[3d38a0f](https://github.com/Arrow-air/svc-storage/pull/59/commits/3d38a0f9d841f41f152b652cac774723a8c9dbe7)